chain/chain/src/resharding.rs

```diff
@@ -181,15 +202,49 @@ impl Chain {
         sync_hash: &CryptoHash,
         state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
     ) -> Result<(), Error> {
-        let prev_hash = *self.get_block_header(sync_hash)?.prev_hash();
+        let state_roots = state_roots?;
+        let block_header = self.get_block_header(sync_hash)?;
+        let prev_hash = block_header.prev_hash();
+
+        let child_shard_uids = state_roots.keys().collect_vec();
```

Comment:
Does it mean the only operation that resharding can do is splitting a shard into multiple shards?

Reply:
For this stage of the project we are only aiming to support splitting. However not all shards need to be split. We are planning to perform a resharding in testnet and mainnet where we only split shard 3 and we end up with 5 shards in total.